### PR TITLE
Save repetitions in CrossEntropyResult and make it into a JSON serializable dataclass

### PR DESCRIPTION
**Pull request by kevinsung**
@XiaoMiQC

### CODE REVIEW

**Comment by kevinsung** [comment]
Mypy error same as #2769

**Comment by kevinsung** [comment]
I just made mypy ignore the error.

File: cirq/experiments/cross_entropy_benchmarking.py [hunk 1]
```diff
@@ -16,33 +16,25 @@
...
-        Each CrossEntropyPair is a NamedTuple that contains a cycle number and
-        the corresponding XEB fidelity.
-        """
-        return self._data
+    """Results from a cross-entropy benchmarking (XEB) experiment.
+
+    Attributes:
+        data: A sequence of NamedTuples, each of which
+            contains two fields: num_cycle which returns the circuit
+            depth as the number of cycles and xeb_fidelity which returns
```
Comment (viathor) [inline cirq/experiments/cross_entropy_benchmarking.py]: nit: "as the number of cycles" explains nothing unless "cycle" itself is defined. I've seen the term being used for both a single layer of gates and for two layers: single-qubit gates following by two-qubit gates. Probably worth commenting on this explicitly.

Comment (kevinsung) [reply cirq/experiments/cross_entropy_benchmarking.py]: Done.

**Pull request merged**